src/cpp/session/modules/SessionSource.cpp

```diff
@@ -86,6 +88,23 @@ void writeDocToJson(boost::shared_ptr<SourceDocument> pDoc,
          LOG_ERROR(error);
    }
    (*pDocJson)["notebook"] = notebook;
+   
+   // discover a project-specific tab width if available
+   FilePath docPath = module_context::resolveAliasedPath(pDoc->path());
+   
+   r_util::RProjectConfig projConfig;
+   Error error = r_util::findProjectConfig(docPath, &projConfig);
```

Comment:
Seems like this will run a lot even for docs in the current project -- any chance this is expensive (NFS, etc.)? If so, can we use the known config of the current project if the doc resides in it?